docs/Llama2FineTuning.md

```diff
@@ -0,0 +1,152 @@
+# Experiment with Llama2 model
+
+The goal of this experiment is to evaluate how well a large language model (LLM) can learn the conversion
+between English and Blissymbolics sentence structures. To leverage the LLM's knowledge of English, Blissymbolics
+sentences are composed using English words while adhering to the grammatical and syntactical rules of Blissymbolics.
+For instance, the English sentence "I slowly move towards the blue lake" would be expressed in Blissymbolics as
+"present: I move slowly towards lake blue". Without delving into the linguistic intricacies of Blissymbolics, it is
+essential to note that the language follows a specific ordering and structure to indicate verb tenses, as well as the
+relationships between verbs and adverbs, and nouns and adjectives.
+
+The experiment uses the [7B parameter Llama2 model pretrained by Meta](https://huggingface.co/meta-llama/Llama-2-7b-hf),
+converted for the seamless use of the Hugging Face Transformers format. This model is choosen as a starting
+point because it requires less training time and GPU resources compared to its larger counterparts, while it
+potentially sacrifies some capability. Additionally, the Hugging Face Transformers format is selected because
+of its extensive community support and standardized APIs.
+
+## Download Llama-2-7b-hf to Cedar
+
+1. Request access to Llama2 models on [the Meta website](https://llama.meta.com/llama-downloads/);
+2. Followed the instructions on [the Hugging Face website](https://huggingface.co/meta-llama/Llama-2-7b-hf)
+to request the access to its Llama2 model;
+3. Request a hugging face access token on [this page](https://huggingface.co/settings/tokens);
+4. Login to the Cedar cluster;
+5. Create a "llama" directory and run these commands to download the model:
+
+```
+mkdir llama2
+cd llama2
+
+// Load git-lfs first for downloading via Git large file storage
+module load git-lfs/3.3.0
+git lfs install
+
+git clone https://{hugging_face_id}:{hugging_face_access_token}@huggingface.co/meta-llama/Llama-2-7b-hf
```

Comment:
Followup:  I still have not done this, partly because I do not want to put another copy of Llama2 on cedar.

Also, I remember that we have a shared directory.  We used it to share our StyleGAN results with David.  It's here:  `~/projects/ctb-whkchun/s2_bliss`.  Is it possible to put the LLama LLM in that directory?  Then we could all share it.

We might have already discussed this and found a reason not to -- did we?

Reply:
> All models I downloaded are placed in ~/projects/ctb-whkchun/s2_bliss_LLMs. This folder name has a suffix "_LLMs". All models in there are available to share.

It's great that they are already in a shared directory.  Thanks.